src/core/core.scale.js

```diff
@@ -537,6 +539,34 @@ module.exports = Element.extend({
 		return rawValue;
 	},
 
+	parseValue: function(rawValue) {
+		var value = +this.getRightValue(rawValue);
+		var min, max;
+		if (helpers.isArray(value)) {
+			min = value[0] <= value[1] ? value[0] : value[1];
+			max = value[0] > value[1] ? value[0] : value[1];
+
+			if (min >= 0 && max > 0) {
+				value = max - min;
+			} else if (min <= 0 && max <= 0) {
+				value = min - max;
+			}
```

Comment:
I agree with value being the distance between `min` and `max` but I don't think this method should mirror it. The controller should decides how to interpret `val`. Also,  I would keep the initial user intent and directly use `array[0]` and `array[1]` since it could represent increase or decrease of the value (start and end). What do you think?

```
if (helpers.isArray(value)) {
    min = Math.min(value[0], value[1]);
    max = Math.max(value[0], value[1]);
    val = value[1] - value[0];
} // ...
```

Let's use `Math.min` / `max` here.

Reply:
ok, min and max from Math functions seems to be ok. but with calculationn of val max - min i think will not work. Consider those two situations - [1,4] and [-1,-4]. The corect val as difference between max and min should 3 and -3 respectively. But if we will use max - min it will give 3 in first case that is ok, but it also give 3 in the second case (-1 - -4 = -1 + 4 = 3) and this is wrong. SO thats why i have put that mirrorin also here. In case of negative values we should use min - max (-4 - -1 = -4 + 1= -3) and max - min for positive values (4 - 1 =3 ). Do you agree? Or there is maybe more clear solution here?

Reply:
Simon has a fair point about maintaining the order. E.g. you could imagine we might want to color `[1, 3]` as green since it's rising and `[3, 1]` as red since it's falling.

I wonder if it'd be more useful to parse `start` and `end` here instead of `min` and `max`. We're given those to start it seems like we should just hold on to them instead of figuring out which is the `min`, which is the `max` and then recomputing the `start` and `end` from those values (e.g. right now I see the code above `start = value.val >= 0 ? value.min : value.max;`)

Reply:
hm, not sure about the order. Is what you are showing true fo bar chart or ts only applicaple to line chart? Cause for bars [1,3] and [3,1] should be treated the same. Because its only for line chart i think you can have different points like 1, 3, 1 and colour lines from 1 to 3 and 3 to 1.  If you have exact thought that we should have reversed bars like [3,1] where top of the bar will be faced down, thats interesting thought but i am not sure it wask a task. for now all bars are plotted with only from 0 to Infinity direction and not from Infinity to 0. Hope, I made it clear )

Reply:
@gwyneblaidd that's not what happen with `val = value[1] - value[0];` because `[1,4] -> 3` and `[-1, -4] -> -3`, which is consistent. drawing reverse bar is maybe not supported now but `parseValue` should be generic enough to allow this kind of implementation without breaking changes.

@benmccann I think it's still good to keep `min` and `max` for scales and avoid duplicating the `Math.min/max` all around. But also computing `start` and `end` could be interesting to avoid checking `val` sign. `start` should be `values[0]` (not necessary `min`) and `end` should be `value[1]` (not necessary `max`).

Since `parseValue` API is still unsure, I would make that method private until we are confident to promote it public. Finally, if we introduce `start` and `end`, I would maybe pick `value` instead of `val` (except if you have a 3 letters alternative to `start` :) ). Here is how I think `parseValue` should be implemented:

```javascript
/**
 * @private
 */
_parseValue: function(raw) {
    var value = +this.getRightValue(raw);
    var start, end;

    if (helpers.isArray(value)) {
        start = value[0];
        end = value[1];
    } else {
        start = 0;
        end = value;
    }

    return {
        min: Math.min(start, end),
        max: Math.max(start, end),
        value: end - start,
        start: start,
        end: end
    };
}
```

Reply:
@simonbrunel ok, lets keep start and end but as you mentioned above reversed bars are not currently supported, so thats only just for future implementation.  Also its confusing to use  this calculations val = value[1] - value[0]. With max and min checks you could use min - max or max - min pair to calculate val, but with value[1] - value[0] it will not work for every case. Consider those 4 bars:

[1,4]
[4,1]
[-1,-4]
[-4,-1]

correct difference for val are:  1st= 3, 2nd= 3, 3d = -3, 4th=-3 if we use switch max - min and min - max

and we get wrong set with value[1] - value[0]:

1st=3, 2d=-3, 3d=-3, 4th=3

Again, we dont currently support reversed bars so here we consider here up-oriented bars :).

Reply:
To me the correct calculation is: `1st=3, 2d=-3, 3d=-3, 4th=3` because 1st and 4th are value increase and 2nd and 3rd are value decrease. It doesn't matter how we handle it in the bar controller, this particular method **must** be generic, future proof and not oriented to any interpretation.

> Again, we dont currently support reversed bars so here we consider **here** up-oriented bars

I think that's a wrong approach, **here** has nothing to do with bar, line, or whatever. It's a scale method that returns standardized values and this should allow **any** kind of implementation. If you want to consider up-oriented bars only, then do that in the bar controller:

```javascript
// bar controller
var data = parseValue(raw);
var value = Math.abs(data.value);
value = data.end < 0 ? -value : value;
// ...
```

Reply:
Since we don't agree about `value`, we may want to remove it from the result of `parseValue` and compute it in the controller instead. I think it's only needed in the bar controller right now, so we can re discuss this point later when will actually need it.

Reply:
yep, thas what i thought. we can keep also perfomance here, cause for setting scale (determineDataLimits) we dont need val (value) from parseValue at all but still calculating this. So, let me do changes

Reply:
Oh, I didn't refresh the page before posting my previous comment (so didn't view yours). So yes, we agree about removing `value` from `parseValue`, so should be the same code as the one in my previous comment, without `value: end - start`.